### PR TITLE
[CLIENT] Fix missing pkg_resources module on Python 3.12

### PR DESCRIPTION
**Pull request by dolfinus**
### Problem

After installing `openlineage-python` on Python 3.12 virtual environment, I try to import it:
```python
from openlineage.client import OpenLineageClient
```

I got an exception:
```
ImportError occurred when trying to import numpy module.
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/home/maxim/Repo/airflow-cd/venv/lib/python3.12/site-packages/openlineage/client/__init__.py", line 5, in <module>
    from openlineage.client.client import OpenLineageClient, OpenLineageClientOptions
  File "/home/maxim/Repo/airflow-cd/venv/lib/python3.12/site-packages/openlineage/client/client.py", line 19, in <module>
    from openlineage.client.transport import Transport, TransportFactory, get_default_factory
  File "/home/maxim/Repo/airflow-cd/venv/lib/python3.12/site-packages/openlineage/client/transport/__init__.py", line 9, in <module>
    from openlineage.client.transport.kafka import KafkaConfig, KafkaTransport
  File "/home/maxim/Repo/airflow-cd/venv/lib/python3.12/site-packages/openlineage/client/transport/kafka.py", line 12, in <module>
    from pkg_resources import parse_version
ModuleNotFoundError: No module named 'pkg_resources'
```

This is because `pkg_resources` is not a part of stdlib, but actually provided by `setuptools`. And setuptools is not automatically installed to virtual environments since Python 3.12.

For example, Python 3.11:
```bash
$ python -V
Python 3.11.8
$ python -m venv venv
$ source venv/bin/activate
$ pip freeze --all
pip==24.0
setuptools==65.5.0
```

Python 3.12:
```bash
$ python -V
Python 3.12.2
$ python -m venv venv
$ source venv/bin/activate
$ pip freeze --all
pip==24.0
```

#### One-line summary:

Remove `pkg_resources` dependency and replace it with [packaging](https://packaging.pypa.io/en/latest/version.html) lib.

### Checklist

- [X] You've [signed-off](https://github.com/OpenLineage/OpenLineage/blob/main/why-the-dco.md) your work
- [X] Your pull request title follows our [guidelines](https://github.com/OpenLineage/OpenLineage/blob/main/CONTRIBUTING.md#creating-pull-requests)
- [ ] Your changes are accompanied by tests (_if relevant_)
- [X] Your change contains a [small diff](https://kurtisnusbaum.medium.com/stacked-diffs-keeping-phabricator-diffs-small-d9964f4dcfa6) and is self-contained
- [ ] You've updated any relevant documentation (_if relevant_)
- [x] Your comment includes a one-liner for the changelog about the specific purpose of the change (_if necessary_)
- [ ] You've versioned the core OpenLineage model or facets according to [SchemaVer](https://docs.snowplowanalytics.com/docs/pipeline-components-and-applications/iglu/common-architecture/schemaver) (_if relevant_)
- [ ] You've added a [header](https://github.com/OpenLineage/OpenLineage/tree/main/.github/header_templates.md) to source files (_if relevant_)

----
SPDX-License-Identifier: Apache-2.0\
Copyright 2018-2023 contributors to the OpenLineage project

### CODE REVIEW

File: integration/airflow/openlineage/airflow/__init__.py [hunk 1]
```diff
@@ -6,21 +6,21 @@
 import logging
 
 from openlineage.airflow.version import __version__
-from pkg_resources import parse_version
+from packaging.version import parse as Version
```
Comment (kacpermuda) [inline integration/airflow/openlineage/airflow/__init__.py:9]:
```suggestion
from packaging.version import Version
```

File: client/python/pyproject.toml [hunk 2]
```diff
@@ -29,6 +29,7 @@ dependencies = [
   "python-dateutil>=2.8.2",
   "pyyaml>=5.4",
   "requests>=2.20.0",
+  "packaging",
```
Comment (kacpermuda) [inline client/python/pyproject.toml:32]: Not sure which one, but we probably shouldn't leave it without some min version ?

**Review by kacpermuda — COMMENTED** [review]
LGTM

File: client/python/pyproject.toml (hunk 2 above)
Comment (dolfinus) [reply client/python/pyproject.toml:32]: I don't know either, `Version` class is present in this package since v14.0.0 which was a first released version

Comment (kacpermuda) [reply client/python/pyproject.toml:32]: @JDarDagran any thoughts ?

Comment (dolfinus) [reply client/python/pyproject.toml:32]: Why there should be a minimal version set in the first place?

Comment (mobuchowski) [reply client/python/pyproject.toml:32]: I would rather pin it to some higher version - less chance that something breaks if we use it in some other way and someone else decides to use very low version.

**Review by mobuchowski — APPROVED** [review]

**Pull request merged**